integrationTests/mock/oneSCExecutorMockVM.go

```diff
@@ -218,6 +222,55 @@ func (vm *OneSCExecutorMockVM) processAddFunc(input *vmcommon.ContractCallInput,
 	}, nil
 }
 
+func (vm *OneSCExecutorMockVM) processWithdrawFunc(input *vmcommon.ContractCallInput, value *big.Int) (*vmcommon.VMOutput, error) {
+
+	destNonce, err := vm.blockchainHook.GetNonce(input.RecipientAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	destBalance, err := vm.blockchainHook.GetBalance(input.RecipientAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	if destBalance.Cmp(value) < 0 {
+		return nil, errInsuficientFunds
+	}
+
+	newSCBalance := big.NewInt(0).Sub(destBalance, value)
+	scOutputAccount := &vmcommon.OutputAccount{
+		Nonce:   destNonce,
+		Balance: newSCBalance,
+		Address: input.RecipientAddr,
+	}
+
+	senderNonce, err := vm.blockchainHook.GetNonce(input.CallerAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	senderOutputAccount := &vmcommon.OutputAccount{
+		Address: input.CallerAddr,
+		//VM does not increment sender's nonce
+		Nonce: senderNonce,
+		//tx succeed, return 0 back to the sender
+		Balance: value,
```

Comment:
The value has not changed. It is still the amount from the transaction, it is not 0. Is it ok?

Reply:
it's ok in this case. OuptutAccount does not have a value field, so when some balance needs to be sent back it will be populated in the Balance field. In this case, `value` represents the value of the parameter of the `Withdraw` function in the smart contract. This output account will be converted by the smart contract processor into a cross shard transaction and the `Balance` field will become the `Value` of the transaction

Reply:
ok